### PR TITLE
Add some use cases to DG

### PR DESCRIPTION
**Pull request by lyeyixian**

### CODE REVIEW

**Review by underthehai — COMMENTED** [review]
LGTM

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -274,30 +274,210 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+* 1a. User is at the top directory.
+    * 1a1. User requests to delete a person.
+    * 1a2. Wanderlust shows an error message.
+
+    Use case ends.
+    
+* 2a. The input command format is invalid.
+    * 2a1. Wanderlust shows an error message.
+    
+      Use case ends.
```
[the remark this answers is not in the document]
Comment (lyeyixian) [reply docs/DeveloperGuide.md:480]: Need, but i thought haishan is also doing this so I left out the edit and show features for her, or I should just do them?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -274,30 +274,210 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+    * 1a1. User requests to add a person.
+    * 1a2. Wanderlust shows an error message.
+
+    Use case ends.
+    
+* 2a. The input command format is invalid.
+    * 2a1. Wanderlust shows an error message.
+    
+      Use case ends.
+
```
[the remark this answers is not in the document]
Comment (lyeyixian) [reply docs/DeveloperGuide.md:392]: Hmmm I don't know.
@jeannetoh99 @underthehai @timjkong

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -274,30 +274,210 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+* 1a. User is at the top directory.
+    * 1a1. User requests to delete a person.
+    * 1a2. Wanderlust shows an error message.
+
+    Use case ends.
+    
+* 2a. The input command format is invalid.
+    * 2a1. Wanderlust shows an error message.
+
```
[the remark this answers is not in the document]
Comment (lyeyixian) [reply docs/DeveloperGuide.md:479]: > 
> 
> Maybe can add the person to be deleted is not found?

You mean the person is not found in the travel plan or you mean the person is missing in the input command?

[the remark this answers is not in the document]
Comment (lyeyixian) [reply docs/DeveloperGuide.md:479]: I will add the case for target not found, but the error message one cause I wanted to make them generic so don't need to be too specific. The implementer of that feature can do whatever they want to the error message. What do you think?

File: docs/DeveloperGuide.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (lyeyixian) [reply docs/DeveloperGuide.md:480]: @underthehai

File: docs/DeveloperGuide.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (underthehai) [reply docs/DeveloperGuide.md:392]: i think depends on the type if we refer duplicate as having the same name. Activity and accommodation may have duplicated names? But for person maybe not. 

Or duplicate meaning all other fields (cost, tag) the same, then I think we should check for duplicates!

File: docs/DeveloperGuide.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (underthehai) [reply docs/DeveloperGuide.md:495]: okies, i can add on!

**Review by jiaweiteo — APPROVED** [review]
LGTM

**Pull request merged**